core/src/main/java/org/mskcc/cbio/oncokb/util/ClinicalTrialsUtils.java

```diff
@@ -0,0 +1,201 @@
+            tumor = oncotreeMapping.get(oncoTreeCode);
+
+            Set<String> nctIDList = tumor.getTrials().stream().map(trial -> trial.getNctId()).collect(Collectors.toSet());
+
+            List<Trial> trialsInfo = new ArrayList<>();
+            for (String nctID : nctIDList) {
+                if (trialsMapping.containsKey(nctID)) {
+                    Trial trial = trialsMapping.get(nctID);
+                    if (trial != null) {
+                        trialsInfo.add(trial);
+                    } else {
+                        System.out.println("We do not have trial info for " + nctID);
+                    }
+                }
+            }
+            tumor.setTrials(trialsInfo);
+        }
+
+        return tumor;
+    }
+
+    public static List<Trial> getTrialByTreatment(List<Trial> trials, String treatment) {
+        List<Trial> res = new ArrayList<>();
+        Set<String> drugsNames = Arrays.stream(treatment.split(",|\\+")).map(item -> item.trim().toLowerCase()).collect(Collectors.toSet());
+        res = getTrialsByDrugName(trials, drugsNames);
+        return res;
+    }
+
+    private static List<Trial> getTrialsByDrugName(List<Trial> trials, Set<String> drugsNames) {
+        List<Trial> res = new ArrayList<>();
+        for (Trial trial : trials) {
+            List<Arm> arms = trial.getArms();
+            if(arms != null && !arms.isEmpty()) {
+                for (Arm arm : trial.getArms()) {
+                    List<Drug> drugs = arm.getDrugs();
+                    if (drugs != null && !drugs.isEmpty()) {
+                        if (arm.getDrugs().stream().map(drug -> drug.getDrugName().toLowerCase()).collect(Collectors.toSet()).containsAll(drugsNames)) {
+                            res.add(trial);
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+        return res;
+    }
+
+    public static List<Trial> getTrialsForSpecialCancerType(SpecialTumorType specialTumorType) {
+        List<Trial> trials = new ArrayList<>();
+        if(specialTumorType == null) return trials;
+
+        TumorType matchedSpecialTumorType = ApplicationContextSingleton.getTumorTypeBo().getBySpecialTumor(specialTumorType);
+        if (matchedSpecialTumorType == null) return trials;
+
+        return new ArrayList<>(getAllTrialsBySpecialTumorType(specialTumorType));
+    }
+
+    public static List<Trial> getTrialsByCancerType(Map<String, Tumor>  oncotreeMapping, Map<String, Trial>  trialsMapping, String cancerType) {
+        List<Trial> trials = new ArrayList<>();
+
+        Set<String> tumorCodesByMainType = new HashSet<>();
+        List<TumorType> allOncoTreeSubtypes = ApplicationContextSingleton.getTumorTypeBo().getAllSubtypes();
+        for (TumorType oncoTreeType : allOncoTreeSubtypes) {
+            if (oncoTreeType.getMainType() != null && cancerType.equalsIgnoreCase(oncoTreeType.getMainType())) {
```

Comment:
cancerType could be null

Reply:
@zhx828 Just to clarify are you referring to the case when cancerType is null therefore we should return an empty list or when oncoTreeType.getMainType() is null?